LoginServer/PipeServer.cs

```diff
@@ -0,0 +1,558 @@
+                getListenThread().Abort();
+            }
+        }
+
+
+        /// <summary>
+        /// When a new client connects, sets the stream and creates a new thread to listen to them.
+        /// </summary>
+        public void ListenForClients()
+        {
+            SECURITY_DESCRIPTOR sd = new SECURITY_DESCRIPTOR();
+
+            // Set the Security Descriptor to be completely permissive. ////////////////////////////////////////////////
+            InitializeSecurityDescriptor(ref sd, SECURITY_DESCRIPTOR_REVISION);
+            SetSecurityDescriptorDacl(ref sd, true, IntPtr.Zero, false);
+
+            IntPtr ptrSD = Marshal.AllocCoTaskMem(Marshal.SizeOf(sd));
+            Marshal.StructureToPtr(sd, ptrSD, false);
+
+            SECURITY_ATTRIBUTES sa = new SECURITY_ATTRIBUTES
+            {
+                nLength = Marshal.SizeOf(sd),
+                lpSecurityDescriptor = ptrSD,
+                bInheritHandle = 1
+            };
+
+            IntPtr ptrSA = Marshal.AllocCoTaskMem(Marshal.SizeOf(sa));
+            Marshal.StructureToPtr(sa, ptrSA, false);
+
+            while (true)
+            {
+                // Create an instance of a named pipe for one client.
+                SafeFileHandle clientHandle = CreateNamedPipe(
+                    PipeNameIs(),
+                    // DUPLEX | FILE_FLAG_OVERLAPPED = 0x00000003 | 0x40000000;
+                    0x40000003,
+                    0,
+                    255,
+                    BUFFER_SIZE,
+                    BUFFER_SIZE,
+                    0,
+                    ptrSA);
+
+                // If a new named pipe can't be created restart loop.
+                if (clientHandle.IsInvalid)
+                {
+                    continue;
+                }
+
+                int success = ConnectNamedPipe(clientHandle, IntPtr.Zero);
+
+                // couldn't connect to client.
+                if (success == 0)
+                {
+                    // close the handle and wait for the next client.
+                    clientHandle.Close();
+                    continue;
+                }
+
+                Client client = new Client
+                {
+                    handle = clientHandle
+                };
+
+                lock (clients)
+                    clients.Add(client);
+
+                client.stream = new FileStream(client.handle, FileAccess.ReadWrite, BUFFER_SIZE, true);
+
+                // Get the login details
+                // send the login details solely to that client.
+                string str = "Your Username,Password are: ";
+                Byte[] login = Utility.ConvertToBytes(str + GetAdminDetails());
+                SendMessageToOne(login, client);
+
+                // Create a new thread to wait for that clients mesages.
+                Thread readThread = new Thread(Read)
+                {
+                    IsBackground = true
+                };
+                readThread.Start(client);
```

Comment:
shouldn't you assign this thread to the client, so you can kill it when the client disconnects?